cmd/oras/internal/option/remote.go

```diff
@@ -205,22 +206,25 @@ func (opts *Remote) authClient(registry string, debug bool) (client *auth.Client
 			return dialer.DialContext
 		}
 	}
+	// default value are derived from http.DefaultTransport
+	baseTransport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: resolveDialContext(&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}),
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		TLSClientConfig:       config,
+	}
```

Comment:
I think you could do this
```suggestion
	baseTransport := http.DefaultTransport.(*http.Transport).Clone()
```

Reply:
This is a good idea, but not doable for now since we customized `DialContext` to support customized DNS (via `--resolve host:port:address`). Took a rough look and we need to refactor [Dialer](https://github.com/oras-project/oras/blob/v1.0.0-rc.1/internal/net/net.go#L25) to make this change possible, which goes beyond the scope of this PR, I think we can skip this change in this PR. Will create an issue for tracking.

Reply:
issue created: https://github.com/oras-project/oras/issues/839

Reply:
Agreed. It looks cleaner using `Clone()`. By applying `Clone()`, we can have the following code.
```suggestion
	baseTransport := http.DefaultTransport.(*http.Transport).Clone()
	baseTransport.DialContext = resolveDialContext(&net.Dialer{
		Timeout:   30 * time.Second,
		KeepAlive: 30 * time.Second,
	}),
	baseTransport.	TLSClientConfig = config
```
However, it is still not optimal.

As @qweeah mentioned, we can refactor `resolveDialContext` and the following `Dialer`

https://github.com/oras-project/oras/blob/32b9129664039274f91a7a896e25953a92a06f32/internal/net/net.go#L25-L28

to accept `DialContext` instead of `*net.Dialer` to make it further optimal in another PR as
```suggestion
	baseTransport := http.DefaultTransport.(*http.Transport).Clone()
	baseTransport.DialContext = resolveDialContext(baseTransport.DialContext),
	baseTransport.	TLSClientConfig = config
```